src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,229 @@
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create car: " + car, e);
+        }
+        insertDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, c.model AS car_model,"
+                + " m.id AS manufacturer_id, m.name AS manufacturer_name,"
+                + " m.country As manufacturer_country"
+                + " FROM cars c"
+                + " INNER JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement =
+                        connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car by id: " + id, e);
+        }
+
+        if (car != null) {
+            car.setDrivers(getDriversList(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model AS car_model,"
+                + " m.id AS manufacturer_id, m.name AS manufacturer_name,"
+                + " m.country As manufacturer_country"
+                + " FROM cars c"
+                + " INNER JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id"
+                + " WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarStatement =
+                        connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get a list of cars from cars table", e);
+        }
+        if (!cars.isEmpty()) {
+            for (Car car : cars) {
+                car.setDrivers(getDriversList(car.getId()));
+            }
+        }
```

Comment:
If cars is empty you will just have zero iterations of cycle:
```suggestion
        for (Car car : cars) {
            car.setDrivers(getDriversList(car.getId()));
        }
```